x-pack/plugins/cases/server/services/attachments/index.ts

```diff
@@ -336,6 +360,41 @@ export class AttachmentService {
     }
   }
 
+  private transformAndDecodeBulkUpdateResponse(
+    res: SavedObjectsBulkUpdateResponse<AttachmentPersistedAttributes>,
+    comments: UpdateArgs[]
+  ): SavedObjectsBulkUpdateResponse<AttachmentTransformedAttributes> {
+    const validatedAttachments: Array<SavedObjectsUpdateResponse<AttachmentTransformedAttributes>> =
+      [];
+
+    for (let i = 0; i < res.saved_objects.length; i++) {
+      const attachment = res.saved_objects[i];
+
+      if (isSOError(attachment)) {
+        // Forcing the type here even though it is an error. The client is responsible for
+        // determining what to do with the errors
+        // TODO: we should fix the return type of this function so that it can return errors
```

Comment:
Unfortunately the `SavedObjectsBulkUpdateResponse` is wrong so we need to cast it at some point somewhere. Similar to `patchCases` in `x-pack/plugins/cases/server/services/cases/index.ts` we had to cast.

Reply:
Yeah, I mean we can create our own type where we force it to be what we want.